presto-main/src/main/java/com/facebook/presto/operator/VariableWidthBlockEncodingBuffers.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.block.AbstractVariableWidthBlock;
+import com.google.common.annotations.VisibleForTesting;
+import io.airlift.slice.SliceOutput;
+import org.openjdk.jol.info.ClassLayout;
+
+import static com.facebook.presto.array.Arrays.ExpansionFactor.LARGE;
+import static com.facebook.presto.array.Arrays.ensureCapacity;
+import static com.facebook.presto.operator.MoreByteArrays.setBytes;
+import static com.facebook.presto.operator.UncheckedByteArrays.setIntUnchecked;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+
+public class VariableWidthBlockEncodingBuffers
+        extends BlockEncodingBuffers
+{
+    @VisibleForTesting
+    public static final int POSITION_SIZE = Integer.BYTES + Byte.BYTES;
```

Comment:
nit: Can this be package private?  -- ditto for other `POSITION_SIZE` IN `BlockEncodingBuffers`